pyroomacoustics/room.py

```diff
@@ -1164,4 +1204,55 @@ def extrude(self, height):
 
         Room.extrude(self, np.array([0., 0., height]))
 
+    @classmethod
+    def inv_sabine(self, t60, room_dim):
+        '''
+        Given desired T60, (shoebox) room dimension and sound speed,
+        computes the absorption coefficient (amplitude) and image source
+        order needed. The speed of sound used is the package wide default
+        (in :py:data:`pyroomacoustics.parameters.constants`).
+
+        Parameters
+        ----------
+        t60: float
+            Desired T60 (time it takes to go from full amplitude to 60 dB decay) in seconds
+        room_dim: list of floats
+            List of length 2 or 3 of the room side lengths
+
+        Returns
+        -------
+        absorption: float
+            The absorption coefficient (in amplitude domain, to be passed to Room constructor)
+        max_order: int
+            The maximum image source order necessary to achieve the desired T60
+        '''
+        c = constants.get('c')
+
+        # Finding image sources up to a maximum order creates a (possibly 3D) diamond
+        # like pile of (reflected) rooms. Now we need to find the image source model order
+        # so that reflections at a distance of at least up to ``c * rt60`` are included.
+        # One possibility is to find the largest sphere (or circle in 2D) that fits in the
+        # diamond. This is what we are doing here.
+        R = []
+        for l1, l2 in  itertools.combinations(room_dim, 2):
+            R.append(l1 * l2 / np.sqrt(l1 ** 2 + l2 ** 2))
+
+        V = np.prod(room_dim)  # volume
+        # surface computation is diff for 2D and 3D
+        if len(room_dim) == 2:
+            S = np.sum(room_dim)
+            sab_coef = 5  # The Sabine's coefficient needs to be adjusted in 2D
```

Comment:
For the 2D case, I've trying to get an intuitive understanding of what these coefficients/constrants should be. Here's what I've been able to dig up so far.

Video tutorial (skipping to the constant part): https://www.youtube.com/watch?v=A8tWMnr5pps&feature=youtu.be&t=11m22s

Written up derivation of Sabine: https://courses.physics.illinois.edu/phys406/sp2017/Lecture_Notes/P406POM_Lecture_Notes/Derivation_of_the_Sabine_Equation.pdf

So in the well-known [3D case](https://en.wikipedia.org/wiki/Reverberation#Sabine_equation) there's this constant factor of `24` which seems to come from a factor `4` which comes computing the surface area of the sphere multiplied by a factor `6` dropping down from `ln(10^-6)` for the `-60` drop.

So for 2D, I think the combined factor should be `2*6=12` where `2` comes from the area of a circle and the `6` still comes from the `-60 dB` drop.

Reply:
Amazing! Thanks for doing the digging! That falls square in the range that seems correct. Ampirically, I had this 5, but after correcting the computation of the perimeter, this should become 10. That is quite close to the correct 12 that you derived. I'll make the change.